From 5d3f9ff8a407e3da1eaad1ed8077b62f53439731 Mon Sep 17 00:00:00 2001
From: Priya N S <priya.ns@ti.com>
Date: Fri, 7 May 2021 19:08:35 +0530
Subject: [PATCH] create-sdcard: Fix wrong rootfs tarball selection

* Validate user input of rootfs tarball selection.
* This will not allow the user to proceed with wrong rootfs
  tarball selection.

Signed-off-by: Priya N S <priya.ns@ti.com>
Signed-off-by: Sekhar Nori <nsekhar@ti.com>
---
 create-sdcard.sh | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/create-sdcard.sh b/create-sdcard.sh
index 264095a7e8f4..8f996871d0fc 100644
--- a/create-sdcard.sh
+++ b/create-sdcard.sh
@@ -767,9 +767,19 @@ cat << EOM

 EOM
		ls --sort=size $ROOTFILEPARTH | grep "tisdk.*image" | grep 'tar.xz' | grep -n '' | awk {'print "	" , $1'}
-		echo ""
-		read -p "Enter Number of rootfs Tarball: " TARNUMBER
-		echo " "
+		COUNT=`ls $ROOTFILEPARTH | grep "tisdk.*image" | grep 'tar.xz' | grep -n '' | awk {'print $1'} | wc -l`
+		ENTERCORRECTLY="0"
+		while [ $ENTERCORRECTLY -ne 1 ]
+		do
+			read -p "Enter Number of rootfs Tarball: " TARNUMBER
+			echo " "
+			if [ -z "${TARNUMBER//[0-$COUNT]}" ] && [ -n "$TARNUMBER" ] ; then
+				ENTERCORRECTLY=1
+			else
+				echo "Invalid selection!"
+			fi
+			echo ""
+		done
		FOUNDTARFILENAME=`ls --sort=size $ROOTFILEPARTH | grep "image" | grep 'tar.xz' | grep -n '' | grep "${TARNUMBER}:" | cut -c3- | awk {'print$1'}`
		ROOTFSTAR=$FOUNDTARFILENAME

--
2.17.1
